app/controllers/submitted_proposals_controller.rb

```diff
@@ -182,12 +182,25 @@ def reviews_excel_booklet
     end
   end
 
+  def proposal_outcome_location
+    proposal_ids = params[:proposal_ids]
+    proposal_ids.each do |id|
+      proposal = Proposal.find_by(id: id)
+      proposal.update(outcome_location_params)
+    end
```

Comment:
This loop is doing 2 queries for every proposal selected. You could make it much more efficient by doing 1 query to get all selected proposals, and then 1 query per proposal to update. Also please use Strong Params.

```
Proposal.find_by(id: selected_proposals_params).each do |proposal|
   proposal.update(outcome_location_params)
end
```

Reply:
Actually, you could have it do 1 update query per proposal submitted by iterating over the posted ids, and then doing an update on that id without fetching the proposal (`UPDATE proposal SET ... WHERE id= id`).

Reply:
Would this work?
```
   Proposal.where(id: proposal_ids).update_all(outcome_location_params)
```